codex/node.nim

```diff
@@ -676,32 +672,45 @@ proc onStore(
 
     return success()
 
-  without indexer =?
-    manifest.verifiableStrategy.init(0, manifest.blocksCount - 1, manifest.numSlots).catch,
-    err:
-    trace "Unable to create indexing strategy from protected manifest", err = err.msg
-    return failure(err)
-
   if slotIdx > int.high.uint64:
     error "Cannot cast slot index to int", slotIndex = slotIdx
     return
 
-  without blksIter =? indexer.getIndicies(slotIdx.int).catch, err:
-    trace "Unable to get indicies from strategy", err = err.msg
-    return failure(err)
+  if isRepairing:
+    trace "start repairing slot", slotIdx
+    try:
+      let erasure = Erasure.new(
+        self.networkStore, leoEncoderProvider, leoDecoderProvider, self.taskpool
+      )
+      if err =? (await erasure.repair(manifest)).errorOption:
```

Comment:
Yeah, because slotIdx is not used to perform repair, this node will end up with the entire dataset stored locally. That's not a problem. (In fact, for the network that's kind of nice!) But it's not what a storage-provider signed up for. They agreed to store a slot. Not the whole thing.

It would be ideal if we could somehow fetch *only those block* we need to repair the slot we are trying to store, and nothing more. I understand that's a complicated puzzle. And for our first try to tackle repair, I entirely support the way you have implemented it here. However, everyone should be aware that it is being implemented this way.